tests/integration/test_lsp_analysis_for_edge_cases.py

```
@@ -0,0 +1,295 @@
+            )
+        )
+
+    def test_source_files(self, analysis: AnalysisRunData):
+        language = analysis.fixture["language"]
+        source_files = analysis.all_results[0].get_source_files(language)
+        source_files_rel = {str(Path(f).relative_to(analysis.project_path.resolve())) for f in source_files}
+        missing = [f for f in analysis.fixture.get("expected_source_files_contain", []) if f not in source_files_rel]
+        assert not missing, (
+            f"Missing {len(missing)} expected source files:\n"
+            + "\n".join(f"  - {f}" for f in missing)
+            + f"\n\nActual files:\n"
```

Ruff (0.15.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)